pkg/audit/service.go

```diff
@@ -39,12 +43,10 @@ func (service *Service) Run(ctx context.Context) (err error) {
 	if err != nil {
 		return err
 	}
-	for _, fail := range failedNodes {
-		service.statdb.RecordFailedAudit(fail)
+	err = service.Reporter.RecordFailedAudits(ctx, failedNodes)
+	// if Error.Has(err) then log the error because it means not all node stats updated
+	if !Error.Has(err) && err != nil {
+		return err
```

Comment:
how are we logging the error here?

Reply:
Ah, I meant to say this was a todo. Good catch, I was thinking maybe we can put a logger on the service struct? But also I'm not sure.

Reply:
I think a logger would be good, but I'd make it a separate PR